cirq-google/cirq_google/engine/stream_manager_test.py

```diff
@@ -0,0 +1,124 @@
+    ):
+        future0 = demux.subscribe(message_id='0')
+        future1 = demux.subscribe(message_id='1')
+        demux.publish(RESPONSE1)
+        demux.publish(RESPONSE0)
+        actual_response0 = await asyncio.wait_for(future0, timeout=1)
+        actual_response1 = await asyncio.wait_for(future1, timeout=1)
+
+        assert actual_response0 == RESPONSE0
+        assert actual_response1 == RESPONSE1
+
+    @pytest.mark.asyncio
+    async def test_message_id_does_not_exist(self, demux):
+        future = demux.subscribe(message_id='0')
+        demux.publish(RESPONSE1)
+
+        assert not future.done()
+
+    @pytest.mark.asyncio
+    async def test_no_subscribers_does_not_throw(self, demux):
+        demux.publish(RESPONSE0)
+
+        # expect no exceptions
+
+    @pytest.mark.asyncio
+    async def test_publishes_twice_for_same_message_id_future_unchanged(self, demux):
+        future = demux.subscribe(message_id='1')
+        demux.publish(RESPONSE1)
+        demux.publish(RESPONSE1_WITH_DIFFERENT_RESULT)
+        actual_response = await asyncio.wait_for(future, timeout=1)
+
+        assert actual_response == RESPONSE1
+
+    @pytest.mark.asyncio
+    async def test_publish_exception_publishes_to_all_subscribers(self, demux):
+        exception = google_exceptions.Aborted('aborted')
+        future0 = demux.subscribe(message_id='0')
+        future1 = demux.subscribe(message_id='1')
+        demux.publish_exception(exception)
+
+        assert future0.exception() == exception
+        assert future1.exception() == exception
+
+    @pytest.mark.asyncio
+    async def test_publish_response_after_publishing_exception_does_not_change_futures(self, demux):
+        exception = google_exceptions.Aborted('aborted')
+        future0 = demux.subscribe(message_id='0')
+        future1 = demux.subscribe(message_id='1')
+        demux.publish_exception(exception)
+        demux.publish(RESPONSE0)
+        demux.publish(RESPONSE1)
+
+        assert future0.exception() == exception
+        assert future1.exception() == exception
+
+    @pytest.mark.asyncio
+    async def test_publish_exception_after_publishing_response_does_not_change_futures(self, demux):
+        exception = google_exceptions.Aborted('aborted')
+        future0 = demux.subscribe(message_id='0')
+        future1 = demux.subscribe(message_id='1')
+        demux.publish(RESPONSE0)
+        demux.publish(RESPONSE1)
+        demux.publish_exception(exception)
+        actual_response0 = await asyncio.wait_for(future0, timeout=1)
+        actual_response1 = await asyncio.wait_for(future1, timeout=1)
+
+        assert actual_response0 == RESPONSE0
+        assert actual_response1 == RESPONSE1
+
+    @pytest.mark.asyncio
+    async def test_unsubscribe_publish_does_not_complete_unsubscribed_future(self, demux):
+        future0 = demux.subscribe(message_id='0')
+        demux.unsubscribe(message_id='0')
+        demux.publish(RESPONSE0)
+
+        assert not future0.done()
```

Comment:
As above, this should be trivially true because the test never awaits its completion.